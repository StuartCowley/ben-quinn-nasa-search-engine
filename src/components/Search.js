/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import "../styles/search.css";
import getImages from "../requests/getImages";

function Search() {
  const [value, setValue] = useState();

  const handleSubmit = (event) => {
    event.preventDefault();
    getImages(value);
  };

  return (
    <div className="search">
      <form className="search__form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="search__input"
          onChange={(event) => setValue(event.target.value)}
        />
        <button className="search__button" type="submit">
          Search
        </button>
      </form>
    </div>
  );
}

export default Search;
