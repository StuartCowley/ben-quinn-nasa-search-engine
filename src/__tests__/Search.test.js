import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("returns button with text", () => {
    const { getByText } = render(<Search />);

    expect(getByText(/search/i)).toHaveAttribute("type", "submit");
  });
});
