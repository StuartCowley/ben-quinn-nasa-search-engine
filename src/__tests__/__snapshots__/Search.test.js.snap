// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search__form"
    >
      <input
        class="search__input"
        type="text"
      />
      <button
        class="search__button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
